bundles/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/internal/model/DataResponseTransformer.java

```diff
@@ -53,27 +64,98 @@ public DataResponseTransformer(ChannelProvider channelProvider) {
                 // This should not happen but we want to get informed about it
                 logger.warn("Channel not found: {}", channelId);
             } else {
-                if (channel instanceof QuantityChannel) {
-                    Unit<?> unit = ((QuantityChannel) channel).getUnit();
-                    double factor = ((ScaledChannel) channel).getFactor();
-                    logger.debug("Channel {} transformed to QuantityType ({}*{} {})", channel.getFQName(), value,
-                            factor, unit.toString());
-                    result.put(channel, new QuantityType<>(value * factor, unit));
-                } else if (channel instanceof SwitchChannel) {
-                    logger.debug("Channel {} transformed to OnOffType ({})", channel.getFQName(), value);
-                    OnOffType mapped = ((SwitchChannel) channel).mapValue(value);
-                    result.put(channel, mapped);
-                } else if (channel instanceof ScaledChannel) {
-                    double factor = ((ScaledChannel) channel).getFactor();
-                    logger.debug("Channel {} transformed to scaled NumberType ({}*{})", channel.getFQName(), value,
-                            factor);
-                    result.put(channel, new DecimalType(value * factor));
-                } else {
-                    logger.debug("Channel {} transformed to NumberType ({})", channel.getFQName(), value);
-                    result.put(channel, new DecimalType(value));
+                ChannelTypeUID typeUID = channel.getChannelTypeUID();
+                String type = typeUID == null ? null : typeUID.getId();
+                type = type == null ? "null" : type;
```

Comment:
```suggestion
                String type = typeUID == null ? "null" : typeUID.getId();
```

Reply:
as the id cannot be null, this should be the same